subsys/bluetooth/services/ots_internal.h

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef BT_GATT_OTS_INTERNAL_H_
+#define BT_GATT_OTS_INTERNAL_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <zephyr/types.h>
+#include "ots_l2cap_internal.h"
+#include "ots_oacp_internal.h"
+#include "ots_olcp_internal.h"
+
+/**@brief OTS Attribute Protocol Application Error codes. */
+enum bt_gatt_ots_att_err_codes {
+	/** An attempt was made to write a value that is invalid or
+	 *  not supported by this Server for a reason other than
+	 *  the attribute permissions.
+	 */
+	BT_GATT_OTS_WRITE_REQUEST_REJECTED	= 0x80,
+	/** An attempt was made to read or write to an Object Metadata
+	 *  characteristic while the Current Object was an Invalid Object.
+	 */
+	BT_GATT_OTS_OBJECT_NOT_SELECTED		= 0x81,
+	/** The Server is unable to service the Read Request or Write Request
+	 *  because it exceeds the concurrency limit of the service.
+	 */
+	BT_GATT_OTS_CONCURRENCY_LIMIT_EXCEEDED	= 0x82,
+	/** The requested object name was rejected because
+	 *  the name was already in use by an existing object on the Server.
+	 */
+	BT_GATT_OTS_OBJECT_NAME_ALREADY_EXISTS	= 0x83,
+};
+
+struct bt_gatt_ots_obj_id {
+	union {
+		uint64_t val;
+		uint8_t ba[6];
+	};
+};
+
+struct bt_gatt_ots_obj_size {
+	uint32_t cur;
+	uint32_t alloc;
+};
+
+struct bt_gatt_ots_obj_metadata {
+	uint64_t id;
+	char *name;
+	struct bt_gatt_ots_obj_type type;
+	struct bt_gatt_ots_obj_size size;
+	uint32_t props;
+};
```

Comment:
I'm planning to get rid of it and use the one from the public header (also rename it as you suggested in the previous comment). Additionally, I will move ``id`` out of it since it is a special type of metadata that is frequently used to reference objects.